docs/content/troubleshooting.md

```diff
@@ -1,113 +1,119 @@
-This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error. To remove the error message, disable Burst compilation by unchecking `Jobs > Enable Burst Compilation`
-in the Unity Editor.
+**Cause**<br/>
 
-<br/>
+This is a benign exception that is thrown when building a worker while burst compilation is turned on. Your worker was successfully built despite this error.
 
-## Log warnings
+**Workaround**<br/>
 
-#### Unity Editor logs warning
+To remove the error message, disable Burst compilation by unchecking `Jobs > Enable Burst Compilation` in the Unity Editor.
 
-The Unity Editor logs a bunch of warnings when I open the SpatialOS GDK Unity project of the  for the first time. 
+## Warnings
 
-**Issue and workarounds**<br/>
-The warnings originate from code written by Unity, e.g. from `com.unity.mathematics` or `com.unity.entities`.
-These warnings are caused by code maintained by Unity over which we have no control. The warnings are benign so you can ignore them.
+#### Several warnings when opening the project for the first time
 
-<br/>
+**Cause**<br/>
+These warnings originate from code written by Unity, `com.unity.mathematics` or `com.unity.entities` for example. These warnings are benign so you can ignore them.
 
-#### Unity Editor logs warning
-Several error messages are logged after I destroyed a GameObject or ECS entity representing a SpatialOS entity.
+#### Several warnings when deleting a GameObject or ECS entity that represents a SpatialOS entity
 
-**Issue and workarounds**<br/>
-Do not proactively destroy GameObjects or ECS entities representing SpatialOS entities. The SpatialOS GDK cleans them up automatically when the respective SpatialOS entity is removed from the worker. Send a [DeleteEntity world command]({{urlRoot}}/content/gameobject/world-commands.md) to delete the entity on the SpatialOS side instead.
+**Cause**<br/>
+These warnings occur when a [GameObject]({{urlRoot}}/content/glossary#gameobject) or [ECS entity]({{urlRoot}}/content/glossary#unity-ecs-entity) that represents a SpatialOS entity is deleted before its corresponding SpatialOS entity leaves the worker's view. The GDK is designed so that the GameObjects and ECS entities that represent SpatialOS entities should only be deleted after the SpatialOS entity they represent has left that worker's view. ECS entities are deleted automatically by the GDK when their corresponding SpatialOS entities leaves that worker's view. In the case of GameObjects, the SpatialOS Runtime sends a message to the Unity worker when a SpatialOS entity leaves that worker's view, the GDK passes this message onto you. It's your responsibility to act upon these commands in your code.
 
-<br/>
+**Fix**<br/>
+When you want to delete a GameObject or ECS entity, send a [DeleteEntity world command]({{urlRoot}}/content/gameobject/world-commands.md) to delete the entity on the SpatialOS side instead.
```

Comment:
Only for deleting GameObjects/ECS entities associated with a SpatialOS entity

Reply:
good callout, added